lib/addons/jupyterhub/index.ts

```diff
@@ -1,23 +1,42 @@
+import * as assert from "assert";
 import { Construct } from "constructs";
 import { ClusterInfo } from '../../spi';
 import { createNamespace, dependable, setPath } from '../../utils';
 import { HelmAddOn, HelmAddOnProps, HelmAddOnUserProps } from '../helm-addon';
 
+import * as cdk from 'aws-cdk-lib';
+import * as efs from 'aws-cdk-lib/aws-efs';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+
+import * as semver from 'semver';
+
 /**
  * Configuration options for the add-on.
  */
 export interface JupyterHubAddOnProps extends HelmAddOnUserProps {
 
     /**
      * Configurations necessary to use EBS as Persistent Volume
-     * Defines storageClass for EBS Volume type, and
-     * capacity for storage capacity
+     * @property {string} storageClass - storage class for the volume
+     * @property {string} capacity - storage capacity (in Mi or Gi)
      */
-    ebsConfig: {
+    ebsConfig?: {
         storageClass: string,
         capacity: string,
     }
 
+    /**
+     * Configuration necessary to use EFS as Persistent Volume
+     * @property {cdk.RemovalPolicy} removalPolicy - Removal Policy for EFS (DESTROY, RETAIN or SNAPSHOT)
+     * @property {string} pvcName - Name of the Volume to be used for PV and PVC
+     * @property {string} capacity - Storage Capacity (in Mi or Gi)
+     */
+    efsConfig?: {
```

Comment:
I am unclear if both efs and ebs configs are defined what happens? Something takes precedence?
It is worth to define this behavior and call it out in the docs.

Reply:
Line 117 does an assert check to make sure that only one is defined. Otherwise we get an error. 

I will make the changes on the documentation to note this behavior.